paasta_tools/kubernetes_tools.py

```diff
@@ -1187,7 +1142,7 @@ def should_run_uwsgi_exporter_sidecar(
                 if autoscaling_params.get(
                     "use_prometheus",
                     DEFAULT_USE_PROMETHEUS_UWSGI
-                    or system_paasta_config.default_should_run_uwsgi_exporter_sidecar(),
+                    or system_paasta_config.default_should_use_uwsgi_exporter(),
```

Comment:
i think we can technically refactor this method to
```
return (
    self.is_autoscaling_enabled()
    and self.get_autoscaling_params()["metrics_provider"] == "uwsgi"
)```

since there's no other way to scale uwsgi services outside of prometheus these days

but this is probably fine for now if we'd rather make a minimal change!